modeling/inference_models/generic_hf_torch.py

```diff
@@ -242,9 +246,17 @@ def new_rebuild_tensor(
         self.patch_embedding()
 
         if utils.koboldai_vars.hascuda:
+            if utils.args.use_ipex:
```

Comment:
is it correct to have this check under the check for `hascuda`? not sure if this is just one of these variables we have where the name no longer makes sense :P

Reply:
It was complaining about koboldai_vars.gpu_device being 0 (cuda). KoboldAI settings commit should fix this too. Will remove.

Reply:
hascuda is used for hasxpu too :P
Probably unnecessary but hascuda is being checked by torch.xpu.is_available() when using --use-ipex.